client/app/components/navigation/Profile.tsx

```diff
@@ -1,32 +1,16 @@
 import Button from "@mui/material/Button/Button";
 import Link from "@mui/material/Link";
-import { signOut, useSession } from "next-auth/react";
-import { getUserFullName } from "@/app/utils/getUserFullName";
-import { useEffect } from "react";
-
+import { signOut } from "next-auth/react";
 import { getEnvValue } from "@/app/utils/actions";
-export default function Profile() {
-  /* use the NextAuth useSession hook to get session data, and if a specific error condition is met,
-     triggers a forced sign-in using the "keycloak" provider to potentially resolve the error related to refreshing access tokens.*/
-  const { data: session } = useSession();
-
-  const name = getUserFullName(session);
-  // 👇️ run function whenever the session object changes e.g. session.error changes
-  useEffect(() => {
-    if (session?.error === "ErrorAccessToken") {
-      // Keycloak has expired; so, call next-auth Signout
-      signOut();
-    }
-  }, [session]);
-
+export default function Profile({ userFullName }: { userFullName: string }) {
```

Comment:
This is a client component, so we pass the name from a server component higher in the tree (rather than using fetch in useEffect)

Reply:
This ticket is about using the name someone entered on the user profile page instead of using the name associated with their IDIR/BCEID. So I don't think we can use the session provider if we want to use the name from the db?

Reply:
Yes, I understand...the suggested changes would allow for "using the name someone entered on the user profile page instead of using the name associated with their IDIR/BCEID" with the added benefit of encapsulation of session data within the auth session provider.
Shall we discuss?